bftengine/src/bcstatetransfer/InMemoryDataStore.cpp

```diff
@@ -124,25 +124,24 @@ void InMemoryDataStore::setIsFetchingState(bool b) { fetching = b; }
 bool InMemoryDataStore::getIsFetchingState() { return fetching; }
 
 void InMemoryDataStore::setCheckpointBeingFetched(const CheckpointDesc& c) {
-  ConcordAssert(checkpointBeingFetched.checkpointNum == 0);
+  ConcordAssert(checkpointBeingFetched_.checkpointNum == 0);
 
-  checkpointBeingFetched = c;
+  checkpointBeingFetched_ = c;
 }
 
 DataStore::CheckpointDesc InMemoryDataStore::getCheckpointBeingFetched() {
-  ConcordAssert(checkpointBeingFetched.checkpointNum != 0);
+  ConcordAssert(checkpointBeingFetched_.checkpointNum != 0);
 
-  return checkpointBeingFetched;
+  return checkpointBeingFetched_;
 }
 
-bool InMemoryDataStore::hasCheckpointBeingFetched() { return (checkpointBeingFetched.checkpointNum != 0); }
+bool InMemoryDataStore::hasCheckpointBeingFetched() { return (checkpointBeingFetched_.checkpointNum != 0); }
 
 void InMemoryDataStore::deleteCheckpointBeingFetched() {
   // TODO(DD): Create a ctor for CheckpointDesc?
   // NOLINTNEXTLINE(bugprone-undefined-memory-manipulation)
-  memset(&checkpointBeingFetched, 0, sizeof(checkpointBeingFetched));
-
-  ConcordAssert(checkpointBeingFetched.checkpointNum == 0);
+  checkpointBeingFetched_.makeZero();
+  ConcordAssert(checkpointBeingFetched_.checkpointNum == 0);
```

Comment:
This one is really redundent, I think it should be remved.
We don't need to assert on makeZero. 
You can unittest it if you like (doesn't make sense as well, right?).
btw, in general, you should use next time ConcordAssertEQ, else when crashing we wont have the needed infor for debug.

Reply:
I removed it and cleaned up the comments above as they were no longer relevant.